### PR TITLE
Securonix Integration

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
In Progress
TODO:
~~outputs - done all but the create cmds and: violation/activity_data~~
~~detailed_description~~ - none. it's just username/password
~~fetch-incidents~~
create-incident
~~perform-action-on-incident~~
create-watchlist
add-entity-to-watchlist

Commands:
`securonix-list-workflows`
`securonix-get-default-assignee-for-workflow`
`securonix-list-possible-threat-actions`
`securonix-list-policies`
`securonix-list-resource-groups`
`securonix-list-users`
`securonix-list-activity-data`
`securonix-list-violation-data`
`securonix-list-incidents`
`securonix-get-incident`
`securonix-get-incident-status`
`securonix-get-incident-workflow`
`securonix-get-incident-available-actions`
`securonix-perform-action-on-incident` - ~~need to validate~~
`securonix-create-incident` - get OK from securonix but not appearing in UI
`securonix-add-comment-to-incident` - ~~need to validate~~
`securonix-list-watchlists`
`securonix-get-watchlist`
`securonix-check-entity-in-watchlist`
`securonix-create-watchlist` - API call fails
`securonix-add-entity-to-watchlist`

## Related Issues
fixes: https://github.com/demisto/etc/issues/16028

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests - only unit tests
- [ ] Documentation - soon
- [x] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/blob/9a87efd0bd145c7c36a2f9ac623a3e3a38d5d963/Integrations/Securonix/Securonix.yml)
- [ ] [CHANGELOG](https://github.com/demisto/content/blob/9a87efd0bd145c7c36a2f9ac623a3e3a38d5d963/Integrations/Securonix/CHANGELOG.md)

### CODE REVIEW

**Comment by bakatzir** [comment]
Screen Shot 2019-12-30 at 9 57 11

Screen Shot 2019-12-30 at 10 05 28
Screen Shot 2019-12-30 at 10 10 32

File: Integrations/Securonix/Securonix.py [hunk 1]
```diff
@@ -0,0 +1,1196 @@
...
+
+    Args:
+        client: Client object with request.
+        args: Usually demisto.args()
+    Returns:
+        Outputs.
+    """
+    watchlist_name = args.get('watchlist_name')
+
+    watchlist = client.create_watchlist_request(watchlist_name)  # TODO - real api call since not working in our env
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.py]: remove comment, move to leftovers issue

File: Integrations/Securonix/Securonix_description.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# TODO
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix_description.md]: add description

File: Integrations/Securonix/Securonix.py [hunk 3]
```diff
@@ -0,0 +1,1196 @@
...
+    action_name = str(args.get('action_name'))
+    resource_name = args.get('resource_name')
+    workflow = args.get('workflow')
+    comment = args.get('comment')
+    employee_id = args.get('employee_id')
+    criticality = args.get('criticality')
+
+    incident = client.create_incident_request(policy_name, resource_group, entity_type, entity_name, action_name,
+                                              resource_name, workflow, comment, employee_id, criticality)
+    incident_info = incident.get('result')  # TODO check that really works. status OK is lying. not visible in UI, NO ID
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.py]: remove comment, move to leftovers issue

File: Integrations/Securonix/Securonix.py [hunk 4]
```diff
@@ -0,0 +1,1196 @@
...
+
+    Args:
+        client: Client object with request.
+        args: Usually demisto.args()
+    Returns:
+        Outputs.
+    """
+    entity_id = args.get('entity_id')
+
+    watchlist = client.check_entity_in_watchlist_request(entity_id)  # TODO - real api call since not working in our env
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.py]: remove comment, move to leftovers issue

File: Integrations/Securonix/Securonix.py [hunk 5]
```diff
@@ -0,0 +1,1196 @@
...
+        fetch_time: From when to fetch if first time, e.g. `3 days`
+        incident_types: Incident statuses to fetch, can be: all, open, closed
+        last_run: Last fetch object.
+
+    Returns:
+        incidents, new last_run
+    """
+    timestamp_format = '%Y-%m-%dT%H:%M:%S.%fZ'
+    if not last_run:  # if first time running
+        new_last_run = {'time': parse_date_range(fetch_time, date_format=timestamp_format)[0]}
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.py]: are you sure parse date range works and you will always have index [0]?
if not, what will be the error?

File: Integrations/Securonix/Securonix.py [hunk 6]
```diff
@@ -0,0 +1,1196 @@
...
+            'securonix-add-comment-to-incident': add_comment_to_incident,
+            'securonix-list-watchlists': list_watchlists,
+            'securonix-get-watchlist': get_watchlist,
+            'securonix-create-watchlist': create_watchlist,
+            'securonix-check-entity-in-watchlist': check_entity_in_watchlist,
+            'securonix-add-entity-to-watchlist': add_entity_to_watchlist
+        }
+        if command == 'fetch-incidents':
+            fetch_time = params.get('fetch_time')
+            incident_types = params.get('incident_types') if 'incident_types' in params else 'opened'
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.py]: seems strange to params here. why not get that them in client init?

File: Integrations/Securonix/Securonix.py [hunk 7]
```diff
@@ -0,0 +1,1196 @@
...
+            incidents, last_run = fetch_incidents(client, fetch_time, incident_types,
+                                                  last_run=demisto.getLastRun())  # type: ignore
+            demisto.incidents(incidents)
+            demisto.setLastRun(last_run)
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.py]: why are we doing this here?

File: Integrations/Securonix/Securonix.yml [hunk 8]
```diff
@@ -0,0 +1,951 @@
...
+    - contextPath: Securonix.ViolationData.Jobstarttime
+      description: Job start time.
+      type: String
+    - contextPath: Securonix.ViolationData.Policyname
+      description: Policy name.
+      type: String
+    - contextPath: Securonix.ViolationData.Resourcename
+      description: Resource name.
+      type: String
+    - contextPath: Securonix.ViolationData.Rg_id
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.yml]: seems not in standard

File: Integrations/Securonix/Securonix.yml [hunk 9]
```diff
@@ -0,0 +1,951 @@
...
+      description: RG ID.
+      type: String
+    - contextPath: Securonix.ViolationData.Rg_name
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.yml]: seems not in standard

File: Integrations/Securonix/Securonix.yml [hunk 10]
```diff
@@ -0,0 +1,951 @@
...
+    - contextPath: Securonix.ViolationData.Timeline
+      description: Timeline.
+      type: String
+    - contextPath: Securonix.ViolationData.Createdate
+      description: Create date.
+      type: String
+    - contextPath: Securonix.ViolationData.Criticality
+      description: Criticality.
+      type: String
+    - contextPath: Securonix.ViolationData.Datasourceid
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.yml]: seems not in standard

File: Integrations/Securonix/Securonix.yml [hunk 11]
```diff
@@ -0,0 +1,951 @@
...
+      secret: false
+    deprecated: false
+    description: Get incident workflow.
+    execution: false
+    name: securonix-get-incident-workflow
+    outputs:
+    - contextPath: Securonix.Incidents.Workflow
+      description: Incident workflow.
+      type: String
+    - contextPath: Securonix.Incidents.IncidentId
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.yml]: seems not in standard

File: Integrations/Securonix/Securonix.py [hunk 12]
```diff
@@ -0,0 +1,1196 @@
...
+        Dbot representation of severity
+    """
+    priority = priority_str.lower()
+    if priority == 'low':
+        return 1
+    if priority == 'medium':
+        return 2
+    if priority == 'high':
+        return 3
+    return 0
```
[the remark this answers is not in the document]
Comment (yuvalbenshalom) [reply Integrations/Securonix/Securonix.py]: well?

**Comment by bakatzir** [comment]
@kirbles19 README is scarce, as we currently do not have access to the product.

**Comment by kirbles19** [comment]
@bakatzir Done.

**Pull request merged**